File: src/routes/api/v1/health.ts
import { createFileRoute } from "@tanstack/react-router";
import { requestId } from "@/lib/pilot/ids";
import { ensureSeeded } from "@/lib/pilot/store";

export const Route = createFileRoute("/api/v1/health")({
  server: {
    handlers: {
      GET: async () => Response.json({ status: "ok", service: "supportpilot-api", request_id: requestId() }),
    },
  },
});

export const readyRoute = {
  GET: async () => {
    await ensureSeeded();
    return Response.json({ status: "ready" });
  },
};

File: backend/app/api/routes/health.py
"""Unauthenticated health checks."""

from fastapi import APIRouter

router = APIRouter()


@router.get("/health")
@router.get("/api/v1/health")
async def health() -> dict[str, str]:
    return {"status": "ok", "service": "supportpilot", "provider": "sandbox"}
